research/sentiment_analysis/data/imdb.py

```diff
@@ -0,0 +1,23 @@
+import tensorflow as tf, numpy as np
+from data.util import to_dataset, pad_sentence, START_CHAR, OOV_CHAR
+
+NUM_CLASS = 2
+_x_train, _y_train, _x_test, _y_test = None, None, None, None
+
+def input_fn(if_training, vocabulary_size, sentence_length, batch_size, repeat=1):
+    global _x_train, _y_train, _x_test, _y_test
```

Comment:
I believe we try to avoid global variables. Can you pass them as arguments?

Reply:
@yhliang2018 (Per the offline discussion) The dataset is probably ~200MB (its compressed format is 17MB) so it can be stored in the memory, whereas I think the proper way to deal with larger data is to keep only a few batches in memory.

In this case, the data is processed given seed_number and word_num, which takes some time (not huge time but relative speaking, as this model does not take very long time to train), so that is why I store it in the global variable.

If you object to global variable, I can run preprocessing once, save the preprocessed data, and then read the data from it instead of doing preprocessing each epoch. @karmel Karmel, can you please provide with what you think on this as well?

Reply:
Many of the other models in the Garden have a preprocessing step before the model runs, from which they save out a file read in by the Dataset. That would be a good way to go here as well.

Reply:
The alternative pattern for these sort of "you only have to calculate it once" variables is something like:
```
def constrict_input_fns():
  _x_train, _y_train, _x_test, _y_test = keras.load_my_dataset()
  def train_input_fn():
    dataset = tf.data.Dataset.from_tensor_slices({"x": _x_train, "y": _y_train})
    ...
    return dataset

  def eval_input_fn():
    dataset = tf.data.Dataset.from_tensor_slices({"x": _x_eval, "y": _y_eval})
    ...
    return dataset

  return train_input_fn, eval_input_fn
```

And then you can call `constrict_input_fns()` and just hang onto the two input fns. I do like the idea that for trivial datasets, you don't have to learn TensorFlow's way to dump to data files because it significantly eases the learning curve.

Reply:
ack